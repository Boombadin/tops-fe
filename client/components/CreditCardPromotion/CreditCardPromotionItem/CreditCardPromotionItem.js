import React from 'react';
import styled from 'styled-components';

import { TextGuide } from '@client/components/Typography';
import withLocales from '@client/hoc/withLocales';
import {
  ELEMENT_ACTION,
  ELEMENT_TYPE,
  generateTestId,
} from '@client/utils/generateElementId';
const CardBox = styled.div`
  flex: 0 0 100%;
`;
const CardBoxInside = styled.div`
  display: flex;
  /* min-height: 48px; */
  height: auto;
  padding: 15px;
  font-size: 13px;
  position: relative;
`;
const CardInfo = styled.div`
  width: 100%;
  display: block;
`;
const CardInfoBank = styled.div`
  display: flex;
  align-items: center;
`;
const CardDefault = styled.div`
  width: 16px;
  height: 16px;
  margin-right: 14px;
`;
const CardDefaultIcon = styled.img`
  width: 16px;
  height: 16px;
`;

const CardTypeIcon = styled.div`
  width: 32px;
  height: 20px;
  display: flex;
  align-items: center;
  margin-right: 17px;
`;
const BankName = styled(TextGuide)`
  color: #171722;
  margin-right: 5px;
`;
const CardDesc = styled.div`
  padding: 10px 0 0 30px;
`;

const CreditCardPromotionItem = ({
  card,
  isSelectedCard,
  lang,
  onSelectCard = () => {},
}) => {
  return (
    <CardBox
      data-testid={generateTestId({
        type: ELEMENT_TYPE.INFO,
        action: ELEMENT_ACTION.VIEW,
        moduleName: 'CreditCardPromotionItem',
        uniqueId: 'Container',
      })}
    >
      <CardBoxInside
        onClick={() => {
          onSelectCard(card?.card_code);
        }}
        data-testid={generateTestId({
          type: ELEMENT_TYPE.BUTTON,
          action: ELEMENT_ACTION.EDIT,
          moduleName: 'CreditCardPromotionItem',
          uniqueId: 'OnSetSelectCard',
        })}
      >
        <CardInfo>
          <CardInfoBank>
            <CardDefault
              data-testid={generateTestId({
                type: ELEMENT_TYPE.IMAGE,
                action: ELEMENT_ACTION.VIEW,
                moduleName: 'CreditCardPromotionItem',
                uniqueId: 'DefaultCard',
              })}
            >
              <CardDefaultIcon
                src={`${
                  isSelectedCard === card?.card_code
                    ? '/assets/icons/red-checked.svg'
                    : '/assets/icons/none-checked.svg'
                }`}
              />
            </CardDefault>
            <CardTypeIcon>
              <img width="100%" src={card?.card_icon} />
            </CardTypeIcon>
            <BankName type="body" bold>
              {card?.card_title?.[lang?.url]}
            </BankName>
          </CardInfoBank>
          <CardDesc
            dangerouslySetInnerHTML={{
              __html: card?.card_promotion_description?.web?.[
                lang?.url
              ]?.replace(/(?:\\r\\n|\\r|\\n)/g, '<br />'),
            }}
          />
        </CardInfo>
      </CardBoxInside>
    </CardBox>
  );
};

export default withLocales(CreditCardPromotionItem);
