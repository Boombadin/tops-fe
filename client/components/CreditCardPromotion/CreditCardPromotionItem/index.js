export { default } from './CreditCardPromotionItem.js';
