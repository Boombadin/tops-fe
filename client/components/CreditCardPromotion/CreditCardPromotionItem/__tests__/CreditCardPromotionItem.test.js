import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import CreditCardPromotionItem from '../CreditCardPromotionItem';
jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});
const TEST_ID = {
  CARD_ITEM: 'inf-viewCreditCardPromotionItem-Container',
  DEFAULT_CARD: 'img-viewCreditCardPromotionItem-DefaultCard',
  SELECT_CARD: 'btn-editCreditCardPromotionItem-OnSetSelectCard',
};

describe('components/CreditCardPromotion/CreditCardPromotionItem', () => {
  const props = {
    card: {
      card_code: 'the1',
    },
    lang: {
      active: true,
      code: 'th_TH',
      name: 'ไทย',
      url: 'th',
    },
    isSelectedCard: {
      card_code: 'the1',
    },
    onSelectCard: () => {},
  };
  test(`it should match with snapshot`, () => {
    const { asFragment } = render(<CreditCardPromotionItem {...props} />);
    expect(
      asFragment(<CreditCardPromotionItem {...props} />),
    ).toMatchSnapshot();
  });

  test('Click selected card | it should be call function onSelectCard()', () => {
    const newProps = {
      ...props,
      card: {
        card_code: 'the1',
      },
      isSelectedCard: {
        card_code: 'the1',
      },
    };
    const { getByTestId } = render(<CreditCardPromotionItem {...newProps} />);
    const setSelectCard = getByTestId(TEST_ID.SELECT_CARD);
    fireEvent.click(setSelectCard);
    expect(setSelectCard).toBeInTheDocument();
  });

  test('Click selected card | it should be show icon', () => {
    const { getByTestId } = render(<CreditCardPromotionItem {...props} />);
    const setDefaultCard = getByTestId(TEST_ID.DEFAULT_CARD);
    expect(setDefaultCard).toBeInTheDocument();
  });
});
