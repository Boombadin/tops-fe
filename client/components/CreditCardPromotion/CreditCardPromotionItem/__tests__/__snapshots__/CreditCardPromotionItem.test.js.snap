// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CreditCardPromotion/CreditCardPromotionItem it should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="CreditCardPromotionItem__CardBox-c1d72l-0 jdQUvJ"
    data-testid="inf-viewCreditCardPromotionItem-Container"
  >
    <div
      class="CreditCardPromotionItem__CardBoxInside-c1d72l-1 ctcRFb"
      data-testid="btn-editCreditCardPromotionItem-OnSetSelectCard"
    >
      <div
        class="CreditCardPromotionItem__CardInfo-c1d72l-2 gnvklv"
      >
        <div
          class="CreditCardPromotionItem__CardInfoBank-c1d72l-3 dgayTT"
        >
          <div
            class="CreditCardPromotionItem__CardDefault-c1d72l-4 kAoDCj"
            data-testid="img-viewCreditCardPromotionItem-DefaultCard"
          >
            <img
              class="CreditCardPromotionItem__CardDefaultIcon-c1d72l-5 ldyXby"
              src="/assets/icons/none-checked.svg"
            />
          </div>
          <div
            class="CreditCardPromotionItem__CardTypeIcon-c1d72l-6 hmsxrY"
          >
            <img
              width="100%"
            />
          </div>
          <p
            class="CreditCardPromotionItem__BankName-c1d72l-7 gfNFKP Text-sc-9p67zt-0 jDYcup"
            color="default"
            display=""
            size="13"
            spacing="-0.4px"
          />
        </div>
        <div
          class="CreditCardPromotionItem__CardDesc-c1d72l-8 kQLqCd"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
